### PR TITLE
Write subtask id to task log header

### PR DESCRIPTION
**Pull request by wrp**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1876

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
Adds a task id to log headers to make it easier to filter logs

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Trivial change to log format

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [ ] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by Fokko** [comment]
Could you make a ticket in Jira? https://issues.apache.org/jira/projects/AIRFLOW This is for the release management and to keep track of why we made the changes.

**Comment by jgao54** [comment]
lgtm~ looks like the error is transient and need a rebuild.

**Comment by bolkedebruin** [comment]
Please uppercase your Jira tag in the commit

**Comment by wrp** [comment]
Ah, sorry, forgot to put the JIRA in the PR description.  Commit subject uppercased.

File: airflow/task_runner/base_task_runner.py [hunk 1]
```diff
@@ -97,7 +97,11 @@ def _read_task_logs(self, stream):
                 line = line.decode('utf-8')
             if len(line) == 0:
                 break
-            self.log.info(u'Subtask: %s', line.rstrip('\n'))
+            self.log.info(
+                u'Subtask %d: %s',
+                self._task_instance.job_id,
```
Comment (jgao54) [inline airflow/task_runner/base_task_runner.py]: @wrp just realized that we typically identify a task by its task_id, so may be better to use `self._task_instance.task_id` here instead. The job_id is an auto-increment primary key of a SchedulerJob/BackfillJob/LocalTaskJob, which for a given dagrun is the same for all tasks.

Comment (wrp) [reply airflow/task_runner/base_task_runner.py]: Good point.  I think it would be helpful to have both in the log header.

**Comment by criccomini** [comment]
LGTM! Merged.

**Pull request closed without merging**